### PR TITLE
feat: add analytics module to handle /analytics endpoints

### PR DESCRIPTION
**Pull request by karrui**
🗒  This PR is complete with the exception of integration tests for AnalyticsRouter, see note below in the test section.

## Problem

This PR extracts out analytics related functions in various controllers and services and reorganises them into the `modules/analytics` directory.

Related to #144

## Solution
**Features**:

- Add static `findFormsWithSubsAbove` method to `SubmissionModel` so it can be easily used by services
- Add static `aggregateFormCount` method to `FormStatisticsTotalModel` so it can be easily used by services
- Add `AnalyticsService` to retrieve user/submission/form counts from their various models.
- Add `AnalyticsFactory` to return correct function handler for retrieving form counts depending on whether `aggregate-stats` feature is enabled in the Feature Manager
- Add `AnalyticsController` controller handler for `/analytics` route
- Replace current `/analytics` route with the new `AnalyticsRouter`

## Tests

- [x] Add unit tests for analytics.service
- [x] Add unit tests for analytics.factory
- [x] Add unit tests for analytics.controller

This will be delayed until https://github.com/opengovsg/FormSG/pull/371 is merged in as that contains code for generated an authenticated session (or should I cherry pick the appropriate commits? Scared that will scare ppl off with the commit size again)
- [ ] Add integration tests for analytics.routes

### CODE REVIEW

File: src/app/modules/analytics/__tests__/analytics.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      const getUserSpy = jest
+        .spyOn(AnalyticsService, 'getUserCount')
+        .mockReturnValueOnce(okAsync(mockUserCount))
+
+      // Act
+      await AnalyticsController.handleGetUserCount(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(getUserSpy).toHaveBeenCalledTimes(1)
+      expect(mockRes.json).toHaveBeenCalledWith(mockUserCount)
```
Comment (liangyuanruo) [inline src/app/modules/analytics/__tests__/analytics.controller.spec.ts:27]: i think you should tighten the test on `mockRes` as it's technically possible to call `res.status(400).json(...)`

Comment (liangyuanruo) [reply src/app/modules/analytics/__tests__/analytics.controller.spec.ts:27]: perhaps something like this?
```suggestion
      expect(mockRes.json).toHaveBeenCalledWith(mockUserCount)
      expect(mockRes.status).not.toHaveBeenCalled()
```

File: src/app/modules/analytics/analytics.service.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+      return new DatabaseError()
+    },
+  )
+}
+
+/**
+ * !!! This function should only be called by {@link AnalyticsFactory} !!!
+ *
+ * ! Access to this function should be determined by whether the `aggregate-stats` feature is enabled.
```
Comment (liangyuanruo) [inline src/app/modules/analytics/analytics.service.ts:64]: consider using the `@private` keyword, although it's still "weird" as the function is exported. perhaps we can introduce dependency injection into the feature manager someday...

**Review by liangyuanruo — APPROVED** [review]
lgtm with minor suggestions!

File: src/app/modules/analytics/__tests__/analytics.controller.spec.ts (hunk 1 above)
Comment (karrui) [reply src/app/modules/analytics/__tests__/analytics.controller.spec.ts:27]: Good point, will update all the missing asserts in this test file

**Pull request merged**